### DISCUSSION TITLE
Configure the property "dev.resteasy.entity.file.threshold" in RESTEasy Client

### DISCUSSION THREAD

**Discussion by Sgitario** [resteasy/resteasy#4085] [Q&A]
We're using RESTEasy Client to generate an API client that has a POST endpoint to upload large files and the client throws an exception because by default RESTEasy Client has a file size limit of 50MB. 
Note that we're using it in a Spring application that does not use Microprofile Config. 
This file size limit is configurable via the property "dev.resteasy.entity.file.threshold" (see the [EntityOutputStream](https://github.com/resteasy/resteasy/blob/770297a2dfada0ab76c367405fe61a55258ca1cb/resteasy-core-spi/src/main/java/org/jboss/resteasy/spi/EntityOutputStream.java#L125) class), however since we're not using Microprofile Config, the only way I found to configure it is via system properties. 
I also tried using the ClientBuilder class as:
```java
clientBuilder.property("dev.resteasy.entity.file.threshold", "500MB");
```
But it didn't make any difference. 
My question is if there is another way to configure this file size limit per client and/or if we can somehow skip this limitation, maybe by using another implementation of EntityOutputStream?

**Comment by jamezp** [comment]
What version of RESTEasy are you using? Setting the property on the client or as a system property should work, if not it's likely a bug we can get fixed.

**Reply by Sgitario** [reply]
We're using the version 6.2.7. 
With the system property works, but it will change for all the clients we have where we can only change one. 
Ideally, this should be configurable using the ClientBuilder:

```
clientBuilder.property("dev.resteasy.entity.file.threshold", "500MB");
```

Do you want me to create an issue to support this?

**Reply by jamezp** [reply]
I'm truly sorry about the long delay on this. I assume given it's been over a year, you've figured something out. It not, please let me know and I will be more than happy to look at this ASAP.

**Comment by LordMaddhi** [comment]
i have the same issue: 
`RESTEASY005081: File limit of 50MB has been reached. The entity cannot be processed. Increase the size with the configuration property dev.resteasy.entity.file.threshold`

i tried adding it to the `application.properties` and to the apiClient with `apiClient.httpClient.property("dev.resteasy.entity.file.threshold", "5G")` but the error still appears. 

any ideas how to fix it?

**Reply by jamezp** [reply]
Wow, I really dropped the ball on this. Let me do some testing and file an issue.

**Reply by yrodriguezl** [reply]
I ran into the same issue, and in my case, what worked was removing the unit from the value.

I was trying to set dev.resteasy.entity.file.threshold=100M, but the error persisted. Then I changed it to dev.resteasy.entity.file.threshold=100000000 (just the number of bytes, without units), and it worked.

Hope this helps!

**Reply by jamezp** [reply]
That is good feedback. Thank you @yrodriguezl
